src/Processors/QueryPlan/Optimizations/optimizeUseAggregateProjection.cpp

```diff
@@ -287,7 +287,7 @@ ActionsDAGPtr analyzeAggregateProjection(
 {
     auto proj_index = buildDAGIndex(*info.before_aggregation);
 
-    MatchedTrees::Matches matches = matchTrees(*info.before_aggregation, *query.dag);
+    MatchedTrees::Matches matches = matchTrees(*info.before_aggregation, *query.dag, false /* check_monotonicity */);
```

Comment:
I do not understand why exactly monotonicity is the issue here.
We use this function for read-in-order analysis, and monotonicity is needed.

Reply:
Answered in Telegram.